gap/oper.gi

```diff
@@ -1239,6 +1239,67 @@ function(digraph, u, v)
   return DIGRAPH_PATH(OutNeighbours(digraph), u, v);
 end);
 
+# DigraphShortestPath
+
+InstallMethod(DigraphShortestPath, "for a digraph and two pos ints",
+[IsDigraph, IsPosInt, IsPosInt],
+function(digraph, u, v)
+  local current, next, parent, distance, falselist, verts, nbs, n, a, b, i, path;
+
+  verts    := DigraphVertices(digraph);
+  nbs      := OutNeighbors(digraph);
+  distance := [];
+
+  if u = v and v in nbs[v] then  # Considers the trivial path from v to v
+    return [[v, v], [Position(nbs[v], v)]];
+  fi;
+
+  #  Setting up objects useful in the function.
+  for i in verts do
+    Add(distance, -1);
+  od;
+  parent  := [];
+  current := [u];
+  next      := [];
+  falselist := [];
+  for i in verts do
+    Add(next, false);
+    Add(falselist, false);
+  od;
+
+  n := 0;  # Counts the loops
+  while current <> [] do
+    n    := n + 1;
+    for a in current do
+        for b in nbs[a] do
+          if distance[b] = -1 then
+            distance[b] := n;
+            next[b]     := true;
+            parent[b]   := a;
+          fi;
+
+          if b = v then
+            path := [[], []];
+            # Finds the path
+            for i in [1 .. n] do
+              Add(path[1], b);
+              Add(path[2], Position(nbs[parent[b]], b));
+              b := parent[b];
+            od;
+            Add(path[1], u);  # Adds the starting vertex to the list of vertices.
+            return [Reversed(path[1]), Reversed(path[2])];
+          fi;
+
+        od;
+      od;
+
+      current := ListBlist(verts, next);
+      next    := IntersectionBlist(next, falselist);
```

Comment:
I think (not 100%) that you'd be better off doing `IntersectBlist(next, falselist);`. The reason being that `IntersectionBlist` produces a new line, whereas `IntersectBlist` changes `next` in-place.